Sources/SwiftProtobuf/ExtensionFields.swift

```diff
@@ -146,12 +143,9 @@ public struct RepeatedExtensionField<T: FieldType>: ExtensionField {
       try T.decodeRepeated(value: &value, from: &decoder)
   }
 
-  public func traverse(visitor: Visitor) throws {
+  public func traverse<V: Visitor>(visitor: inout V) throws {
     if value.count > 0 {
-      try visitor.visitRepeatedField(
-        fieldType: T.self,
-        value: value,
-        fieldNumber: protobufExtension.fieldNumber)
+      try T.visitRepeated(value: value, fieldNumber: protobufExtension.fieldNumber, with: &visitor)
```

Comment:
`visitSingular`, `visitRepeated`, and `visitPacked` here are the new general-purpose reflectors that just dispatch to the appropriate `visitor.visit<modifier><type>Field()` method.